modules/users/server/config/strategies/facebook.js

```diff
@@ -36,6 +37,18 @@ module.exports = function (config) {
 
       // Save the user OAuth profile
       users.saveOAuthUserProfile(req, providerUserProfile, done);
+
+      function generateUsername(profile) {
+        var username = '';
+
+        if (profile.emails) {
+          username = profile.emails[0].value.split('@')[0];
+        } else if (profile.name) {
+          username = profile.name.givenName[0] + profile.name.familyName;
+        }
+
+        return username.toLowerCase() || undefined;
```

Comment:
What happens if on 45 or 47 the username field:
1. does not exist? meaning it's empty for some reason...
2. does not fit to go through validation? for example, it's length is too short/long or uses special UTF8 chars
3. already matches an existing username and then we'll fail on duplicate usernames

If we end up returning undefined, how would the call later to saveOauthUserProfile behave?